addons/account/wizard/account_payment_register.py

```diff
@@ -273,19 +281,36 @@ def _compute_journal_id(self):
     def _compute_available_partner_bank_ids(self):
         for wizard in self:
             if wizard.can_edit_wizard:
-                batches = wizard._get_batches()
-                bank_partners = batches[0]['lines'].move_id.bank_partner_id
+                batch = wizard._get_batches()[0]
+
+                if batch['key_values']['move_type'] in ('sale', 'purchase'):
+                    # For invoices, use `bank_partner_id` that is already handling the difference of logic between
+                    # paying a customer invoice (you are receiving the money on your own bank account) and paying a
+                    # bill (the vendor is receiving the money).
+                    bank_partners = batch['lines'].move_id.bank_partner_id
+                else:
+                    # For journal entries being not an invoice (like reimbursing an employee using an expense), the
+                    # bank account should be retrieved from the partner set on the journal items.
+                    bank_partners = batch['lines'].partner_id
+
                 wizard.available_partner_bank_ids = bank_partners.bank_ids\
-                    .filtered(lambda x: x.company_id in (False, wizard.company_id))._origin
+                    .filtered(lambda x: x.company_id.id in (False, wizard.company_id.id))._origin
             else:
                 wizard.available_partner_bank_ids = False
 
-    @api.depends('available_partner_bank_ids')
+    @api.depends('journal_id', 'available_partner_bank_ids')
     def _compute_partner_bank_id(self):
         for wizard in self:
             if wizard.can_edit_wizard:
-                batches = wizard._get_batches()
-                wizard.partner_bank_id = self.env['res.partner.bank'].browse(batches[0]['key_values']['partner_bank_id'])
+                batch = wizard._get_batches()[0]
+                if batch['key_values']['move_type'] == 'sale' and wizard.journal_id.bank_account_id:
+                    wizard.partner_bank_id = wizard.journal_id.bank_account_id
+                else:
+                    batch_bank_account = self.env['res.partner.bank'].browse(batch['key_values']['partner_bank_id'])
```

Comment:
Bug:
- Create 2 bank accounts A and B for your company, linked to journals J1 and J2
- Create a vendor refund and change the default recipient account (A becomes B)
- Post, click on "register payment"
- The bank account in the wizard is A. It should be B
-----------------------------------------------------------------------------

=> Probably finds its origin around here (not sure)